bciers/apps/reporting/src/app/components/changeReview/templates/facilityReportParser.ts

```diff
@@ -0,0 +1,325 @@
+import { ChangeItem } from "../constants/types";
+
+export function parseFacilityReportField(field: string): {
+  facilityName: string;
+  section: string;
+  activityName?: string;
+  sourceTypeName?: string;
+  unitIndex?: number;
+  fuelIndex?: number;
+  emissionIndex?: number;
+  fieldKey: string;
+  isFacilityLevel?: boolean;
+} | null {
+  // Updated regex to handle facility names with spaces and special characters
+  const facilityMatch = field.match(/\['facility_reports'\]\[(.*?)\]/);
+  if (!facilityMatch) return null;
+
+  // Remove any surrounding quotes from facility name
+  const facilityName = facilityMatch[1].replace(/(^['"])|(['"]$)/g, "");
+
+  const remainingPath = field.substring(
+    facilityMatch.index! + facilityMatch[0].length,
+  );
+
+  const pathSegments = remainingPath.match(/\[[^\[\]]*\]/g) || [];
+  const cleanSegments = pathSegments.map((segment) => {
+    const match = segment.match(/^\[(?:'|")?(.*?)(?:'|")?\]$/);
+    return match ? match[1] : "";
+  });
+
+  if (cleanSegments.length === 0) {
+    return {
+      facilityName,
+      section: "facility",
+      fieldKey: "facility_report",
+      isFacilityLevel: true,
+    };
+  }
+
+  const mainSection = cleanSegments[0];
+
+  if (mainSection === "activity_data") {
+    const activityName = cleanSegments[1];
+
+    if (cleanSegments.length === 2) {
+      return {
+        facilityName,
+        section: "activity_data",
+        activityName,
+        fieldKey: "activity_data",
+      };
+    }
+
+    if (cleanSegments[2] === "source_types") {
+      const sourceTypeName = cleanSegments[3];
+      let unitIndex: number | undefined;
+      let fuelIndex: number | undefined;
+      let emissionIndex: number | undefined;
+      let fieldKey = "";
+
+      for (let i = 4; i < cleanSegments.length; i++) {
+        const segment = cleanSegments[i];
+        const nextSegment = cleanSegments[i + 1];
+
+        if (segment === "units" && nextSegment && /^\d+$/.test(nextSegment)) {
+          unitIndex = parseInt(nextSegment);
+          i++;
+        } else if (
+          segment === "fuels" &&
+          nextSegment &&
+          /^\d+$/.test(nextSegment)
+        ) {
+          fuelIndex = parseInt(nextSegment);
+          i++;
+        } else if (
+          segment === "emissions" &&
+          nextSegment &&
+          /^\d+$/.test(nextSegment)
+        ) {
+          emissionIndex = parseInt(nextSegment);
+          i++;
+        } else if (i === cleanSegments.length - 1) {
+          fieldKey = segment;
+        } else if (i === cleanSegments.length - 2 && cleanSegments[i + 1]) {
+          fieldKey = cleanSegments[i + 1];
+          break;
```

Comment:
I also think having comments in this section would be really helpful - it's not immediately obvious what this chunk of code is doing